code/datums/components/storage/storage.dm

```diff
@@ -115,6 +119,21 @@
 /datum/component/storage/PreTransfer()
 	update_actions()
 
+/datum/component/storage/proc/set_holdable(can_hold_list, cant_hold_list)
+	can_hold_description = generate_hold_desc(can_hold_list)
+
+	can_hold = typecacheof(can_hold_list)
+	cant_hold = typecacheof(cant_hold_list)
+
+/datum/component/storage/proc/generate_hold_desc(can_hold_list)
+	var/desc = ""
+
+	for(var/valid_obj in can_hold_list)
+		var/obj/item/valid_item = new valid_obj()
+		desc += "\n\t <span class='notice'>\a [valid_item.name]</span>"

```

Comment:
Making a new item of every single kind provided in the list is a great way to have strange bugs. You can use initial to get uninitialized values from a type.

Also, not that it matters, you can write this so you only need one span class tag

```R
var/list/desc = list()
for(var/valid_type in can_hold_list)
  var/obj/item/valid_item = valid_type
  desc += "\a [initial(valid_item.name)]"
return "\n\t<span class='notice'>[desc.Join("\n\t")]</span>"